iothub/service/src/MessageSystemPropertyNames.cs

```diff
@@ -8,40 +8,122 @@ namespace Microsoft.Azure.Devices
+        /// </summary>
         public const string CorrelationId = "correlation-id";
 
+        /// <summary>
+        /// An Id used to specify the origin of messages. When messages are generated by IoT Hub, it is set to name of the IoT hub.
+        /// </summary>
         public const string UserId = "user-id";
 
+        /// <summary>
+        /// IoT hub operation
+        /// </summary>
         public const string Operation = "iothub-operation";
 
+        /// <summary>
+        /// A feedback message generator. This property is used in cloud-to-device messages to request IoT hub to generate feedback messages as a result of the consumption of the message by the device.
+        /// </summary>
+        /// <remarks>
+        /// Possible values:
+        /// <list type="bullet">
+        /// <item>
+        /// <description>none (default): no feedback message is generated.</description>
+        /// </item>
+        /// <item>
+        /// <description>positive: receive a feedback message if the message was completed.</description>
+        /// </item>
+        /// <item>
+        /// <description>negative: receive a feedback message if the message expired (or maximum delivery count was reached) without being completed by the device.</description>
+        /// </item>
+        /// <item>
+        /// <description>full: both positive and negative.</description>
+        /// </item>
+        /// </list>
+        /// </remarks>
         public const string Ack = "iothub-ack";
 
+        /// <summary>
+        /// Specifies the device Id from which this message was sent, if there is one.
+        /// </summary>
         public const string ConnectionDeviceId = "iothub-connection-device-id";
 
+        /// <summary>
+        /// The device generation Id of the target device of the cloud-to-device message.
+        /// DeviceGenerationId is an IoT hub generated, case-sensitive string. This value is used to distinguish devices with the
+        /// same device Id when they have been deleted and re-created.
+        /// </summary>
         public const string ConnectionDeviceGenerationId = "iothub-connection-auth-generation-id";
 
+        /// <summary>
+        /// Connection authentication information.
+        /// </summary>
         public const string ConnectionAuthMethod = "iothub-connection-auth-method";
 
+        /// <summary>
+        /// Used to specify the schema of the message content.
+        /// </summary>
         public const string MessageSchema = "iothub-message-schema";
 
+        /// <summary>
+        /// Custom date property set by the originator of the message.
+        /// </summary>
         public const string CreationTimeUtc = "iothub-creation-time-utc";
 
+        /// <summary>
+        /// Used to specify the content encoding type of the message.
+        /// Possible values are utf-8, utf-16, utf-32
```

Comment:
I'd suggest at least a period, and perhaps a colon.

`Possible values are: utf-8, utf-16, and utf-32.`

Reply:
Same elsewhere.